### ISSUE TITLE
Configuring OvenMediaEngine to stream LL-HLS with RTSP input

### ISSUE THREAD

**Issue by deekshithkv** [AirenSoft/OvenMediaEngine#1046]
Hi,
I have installed and build OvenMediaEngine in ubuntu 20.04 as per document provided. I have used RTSP stream from camera as input (RTSP pull), configured OME to serve LL-HLS and OvenPlayer to test the LL-HLS stream. The issue is while loading the video in the player i see a initial delay of around 15-20 sec before playback begins. Is this expected? I am attaching a Wireshark screenshot it shows that for the 1st GET request OME takes around 10-15sec to respond. Can you please provide some details why this delay happens or way to reduce the same?
![Capture](https://user-images.githubusercontent.com/48976953/221591421-0a8119ea-6cd6-4ce6-9b6a-dbcddfa1c51b.PNG)

**Comment by getroot** [comment]
1. LL-HLS can be delivered only after creating a segment. Theoretically, since it is LL-HLS, it is possible to start playback even if only about 3 partial segments are created, but the player often malfunctions if it responds before the segments are created. So OME will start responding once at least 1 segment has been created.

2. Try reducing the SegmentDuration to 2 or 3 seconds to make segments faster.

```
<Publishers>
	<LLHLS>
		<ChunkDuration>0.5</ChunkDuration>
		<SegmentDuration>2</SegmentDuration>
		<SegmentCount>10</SegmentCount>
```

4. Nevertheless, if the segment is created late, check the keyframe interval of the input stream. HLS segments must start with a keyframe. Therefore, the longer the keyframe interval, the longer it takes to create segments.

5. For this reason, it is recommended to create an LL-HLS stream in advance and start playing it. Since the player plays the last partial segment, the above reason does not add any delay.

6. Try playing with WebRTC. We'll start right away.

**Comment by deekshithkv** [comment]
thank you very much for your response. It is really helpful. One more issue we faced is that the latency gradually increases with time. Is there any reported issue? or is there a way to fix it?

**Comment by naanlizard** [comment]
Generally speaking, I see latency grow with RTMP input if the encoder can't keep up. Is that possibly what's happening with your RTSP source?

**Comment by getroot** [comment]
Bugs that OME increase latency have not been reported yet. However, as @naanlizard said, the problem of input stream (encoder, network) was the most common cause. Or there was a cause of the performance of the server that the OME worked. Or because the player's network is slow, latency continues to increase.

Of course, there may be many causes. So if you want to accurately analyze your problem, upload the information required by the Template of this document (server.xml, ovenmediaengine.log file, player information, encoder information).

**Comment by deekshithkv** [comment]
Thank you for your reply, we will look in to it soon. One more issue we faced is that OME is waiting 5sec for RTCP, 
Log: [2023-02-28 03:55:51.256] W [StreamMotor:3663] RtspcStream | rtspc_stream.cpp:984  | Since the RTCP SR was not received within 5 seconds, the PTS is calculated for each track without RTCP SR. (Lip-Sync may be out of sync)
We only send video from  camera and no RTCP packet is send, so is there a way to avoid/configure this 5sec wait and reduce initial playback delay.

**Comment by getroot** [comment]
OvenMediaEngine very strongly recommends that clients send RTCP SRs. There are probably many smart ways to handle exceptions, but such tasks are usually very low priority to me.
